lib/srv/desktop/windows_server.go

```diff
@@ -754,14 +754,15 @@ func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
 	log.Debug("Connecting to Windows desktop")
 	defer log.Debug("Windows desktop disconnected")
 
-	if err := s.connectRDP(ctx, log, tdpConn, desktop, authContext); err != nil {
+	if err := s.connectRDP(ctx, tdpConn, desktop, authContext); err != nil {
 		log.Errorf("RDP connection failed: %v", err)
 		sendTDPError("RDP connection failed.")
 		return
 	}
 }
 
-func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger, tdpConn *tdp.Conn, desktop types.WindowsDesktop, authCtx *auth.Context) error {
+func (s *WindowsService) connectRDP(ctx context.Context, tdpConn *tdp.Conn, desktop types.WindowsDesktop, authCtx *auth.Context) error {
+	log := s.cfg.Log
```

Comment:
This looks a bit hacky as you're shadowing the package name. The implementation below also uses `log` or `s.cfg.Log`. Can we agree on one alias for the logger and either use `s.cfg.Log` or rename log to indicate that it's not the global logger? You could also embed logger into `WindowsService` to use `s.Infof(...)`.

Reply:
I thought I'd found a place where we were unnecessarily passing the logger through functions when we had it right on the `*WindowsService`, but now that I'm looking more closely I realize that we're actually adding fields to the logger [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L710) and [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L736) and [here](https://github.com/gravitational/teleport/blob/27c57f9a76a813fa943465a8d4e9c2bc17e12d4e/lib/srv/desktop/windows_server.go#L753), which is why we were passing it through. Basically `s.cfg.Log` is the global service logger, and then we derive a connection-specific log from it per connection. I've changed the logic back to reflect that in 6f3e479cdc127e1ab75eb667539b403c9e0afd59 

> This looks a bit hacky as you're shadowing the package name.

Unless I'm misunderstanding you, I don't think that's actually the case here. The package name is `logrus` and we're calling this thing `log`.